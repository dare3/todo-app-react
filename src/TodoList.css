/* TodoList.css */
.todo-list {
  margin: 20px 0;
}

.todo {
  display: flex;
  justify-content: space-between;
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 3px;
  margin-bottom: 10px;
}
