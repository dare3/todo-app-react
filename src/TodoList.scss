.todo-list {
    margin: 20px 0;
  }
  
  .todo {
    display: flex;
    justify-content: space-between;
    padding: 10px;
    border: 1px solid #eb8107;
    border-radius: 3px;
    margin-bottom: 10px;
    transition: opacity 0.5s ease;
  
    &.completed {
      text-decoration: line-through;
      color: rgba(124, 33, 33, 0.473);
    }
  
    input {
      margin-right: 10px;
    }
  
    button {
      margin-left: 5px;
      cursor: pointer;
      border: none;
      background-color: #bec5ce;
      color: rgba(6, 10, 236, 0.973);
      border-radius: 3px;
      padding: 5px 10px;
      
      &:hover {
        background-color: #0057b33d;
      }
    }
  }
  
  .fade-out {
    opacity: 0;
  }
  