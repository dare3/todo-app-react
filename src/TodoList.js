// TodoList.js
import React, { useState, useEffect } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import './TodoList.scss'; // Optional for styling

function TodoList() {
  const [todos, setTodos] = useState([]);

  useEffect(() => {
    const storedTodos = JSON.parse(localStorage.getItem('todos')) || [];
    setTodos(storedTodos);
  }, []);

  useEffect(() => {
    localStorage.setItem('todos', JSON.stringify(todos));
  }, [todos]);

  const addTodo = (task) => {
    setTodos([...todos, { task, id: Date.now(), completed: false }]);
  };

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  };

  const updateTodo = (id, newTask, completed) => {
    setTodos(todos.map(todo =>
      todo.id === id ? { ...todo, task: newTask || todo.task, completed: completed !== undefined ? completed : todo.completed } : todo
    ));
  };

  return (
    <div className="todo-list">
      <NewTodoForm addTodo={addTodo} />
      <div>
        {todos.map(todo => (
          <Todo 
            key={todo.id} 
            id={todo.id} 
            task={todo.task} 
            completed={todo.completed}
            removeTodo={removeTodo} 
            updateTodo={updateTodo} 
          />
        ))}
      </div>
    </div>
  );
}

export default TodoList;
