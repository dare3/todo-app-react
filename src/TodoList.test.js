import React from 'react';
import { render, userEvent } from '@testing-library/react';
import TodoList from './TodoList';

test('renders TodoList component', () => {
  const { getByText } = render(<TodoList />);
  expect(getByText(/New Todo/i)).toBeInTheDocument();
});

test('matches snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});



test('adds a new todo', () => {
  const { getByPlaceholderText, getByText } = render(<TodoList />);
  
  const input = getByPlaceholderText(/New Todo/i);
  userEvent.type(input, 'New Todo');
  userEvent.click(getByText(/Add Todo/i));

  expect(getByText(/New Todo/i)).toBeInTheDocument();
});

test('removes a todo', () => {
  const { getByPlaceholderText, getByText } = render(<TodoList />);
  
  const input = getByPlaceholderText(/New Todo/i);
  userEvent.type(input, 'Todo to remove');
  userEvent.click(getByText(/Add Todo/i));

  userEvent.click(getByText(/X/i));

  expect(getByText(/Todo to remove/i)).not.toBeInTheDocument();
});
