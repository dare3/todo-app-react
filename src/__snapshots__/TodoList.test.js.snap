// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-list"
  >
    <form>
      <input
        placeholder="New Todo"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
